### PR TITLE
[Semantic API] Suggest Function Arguments in Expression-bodied Functions

### LINKED ISSUE #31197: [SemanticAPI] function arguments are not suggested in expression function body context
**Description:**
$subject depicted below. 
iss

**Steps to reproduce:**

```ballerina
function func2(string mystr) returns  string => m<cursor>
```

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #31197

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java [hunk 1]
```diff
@@ -231,7 +231,12 @@ public void visit(BLangService serviceNode) {
 
     @Override
     public void visit(BLangFunction funcNode) {
-        if (PositionUtil.withinBlock(this.linePosition, funcNode.getPosition())) {
+        // If the function is Expression-bodied, then a right-inclusive position lookup would be performed.
+        // TODO: Update this approach accordingly once the discussion
+        //  at: https://github.com/ballerina-platform/ballerina-lang/discussions/28983 is concluded
+        if ((funcNode.getBody() instanceof BLangExprFunctionBody &&
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java:237]: Can check the node kind here

Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java]: Updated.

**Comment by mohanvive** [comment]
@sanjana There are some LS test failures. Can you please check and fix.

**Comment by sanjana** [comment]
> @sanjana There are some LS test failures. Can you please check and fix.

@mohanvive fixed the LS test case failures.

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**